adapters/beachfront/beachfronttest/exemplary/adm-video-app.json

```diff
@@ -0,0 +1,124 @@
+{
+  "mockBidRequest": {
+    "id": "adm-video",
+    "imp": [
+      {
+        "id": "video1",
+        "ext": {
+          "bidder": {
+            "bidfloor": 3.01,
+            "appId": "videoAppId1"
+          }
+        },
+        "video": {
+          "mimes": [
+            "video/mp4"
+          ],
+          "context": "instream",
+          "w": 300,
+          "h": 250
+        }
+      }
+    ],
+    "app": {
+      "bundle": "some.domain.us/some/page.html"
+    },
+    "device":{
+      "ip":"192.168.168.168"
+    }
+  },
+
+  "httpCalls": [
+    {
+      "expectedRequest": {
+        "uri": "https://qa.beachrtb.com/bid.json?exchange_id=videoAppId1",
+        "body": {
+          "id": "adm-video",
+          "imp": [
+            {
+              "video": {
+                "w": 300,
+                "h": 250,
+                "mimes": [
+                  "video/mp4"
+                ]
+              },
+              "bidfloor": 3.01,
+              "id": "video1",
+              "secure": 0
+            }
+          ],
+          "app": {
+            "bundle": "some.domain.us/some/page.html",
+            "domain": "domain.some"
```

Comment:
@muncha is `appCopy.Domain = fmt.Sprintf("%s.%s", chunks[len(chunks)-(len(chunks)-1)], chunks[0])` (`adapters/beachfront/beachfront.go` line 453) correct? Shouldn't the domain here be `"domain.us"` instead?

Reply:
@guscarreon @mansinahar 
Yes it should. Looks like a copy paste error.
LGTM - thanks for catching that.

Reply:
@muncha line 53 below gets asserted against your adapter's `MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo)`. Therefore, if we change `adapters/beachfront/beachfronttest/exemplary/adm-video-app.json`'s line 53 to `"domain.us"`, the test will fail. This issue should be addressed by correcting `adapters/beachfront/beachfront.go`'s line 453 shown below:
```
447         if bfReqs[i].Request.App != nil && bfReqs[i].Request.App.Domain == "" && bfReqs[i].Request.App.Bundle != "" {
448             if bfReqs[i].Request.App.Bundle != "" {
449                 var chunks = strings.Split(strings.Trim(bfReqs[i].Request.App.Bundle, "_"), ".")
450
451                 if len(chunks) > 1 {
452                     appCopy := *bfReqs[i].Request.App
453                     appCopy.Domain = fmt.Sprintf("%s.%s", chunks[len(chunks)-(len(chunks)-1)], chunks[0])
454                     bfReqs[i].Request.App = &appCopy
455                 }
456             }
457         }
adapters/beachfront/beachfront.go
```
My team mate @mansinahar is writting the correction and will be included as part of this pull request for you to review and approve

Reply:
@muncha I am guessing this needs to be something like what you do for `site` on L442, right? The `getDomain` method strips out the protocol and the path. If it is, then I can update the PR to use the `getDomain` method for app bundle too. 

Also, if it would be easier for you to push a commit on this PR for this then please feel free to do so. Just lemme know.

Reply:
Hi all, sorry for the delay. I'm pulling this branch and working on it now. To specifically answer @mansinahar - 
"the current code resolves the app bundle “some.domain.us/some/page.html” to app domain as “domain.some”, which doesn’t seem right to me"

Nope, not right. It should be domain.us.

Reply:
Perfect! Thank you so much @muncha. Feel free to push a commit on this PR or close this and open a new one that addresses both the data race and the app domain resolution. Let us know what works best for you :)

Reply:
@mansinahar
I'd rather push to this PR. I'm not sure I know how to do that, though. I've got your fork as a remote, and I've got beachfront-data-race cloned. So, I fix it, then what? Do I push it to /beachfront/prebid-server with the same branch name and there is a hook that will alert this PR? Or more to it than that?

Reply:
I believe you should just be able to push it to this branch on my remote doing something like:
```
git push <my-remote-name> <local-branch-name>:<beachfront-data-race>
```
In case you run into issues with permissions and such then lemme know

Reply:
@mansinahar 
In that test I'm sending a page url in for App.Bundle, but that's not what a bundle is supposed to be:
(/go/pkg/mod/github.com/mxm!cherry/openrtb/v15@v15.0.0/openrtb2/app.go)

	// Description:
	//   A platform-specific application identifier intended to be
	//   unique to the app and independent of the exchange. On
	//   Android, this should be a bundle or package name (e.g.,
	//   **com.foo.mygame**). On iOS, it is typically a numeric ID
	Bundle string `json:"bundle,omitempty"

So really, the test is bad, not the Bundle processing:
- If I send it :  `{ "bundle": "us.domain.some" }` the domain is set to "domain.us". 
- If I send a numeric ID is passes through unharmed, which is what it should do.
- If I send it `{ "bundle": "some.domain.us/some/page.html"}, GIGO, "domain.some"
  - To check for that, I would need to check if
    - The passed bundle starts with a TLD, therefore a valid android bundle. Maybe just .com and .org
    - Or the passed bundle value matches a regex something like /^id[0-9]+/ therefore a valid iOS bundle

I'm not sure that I should check for that, though. What would a windows phone send? I'm sure I could find out, but I think that users would know the correct bundle format to send, and if they send garbage, they get garbage.

Does that seem resonable?

Also - permission denied on your remote.

Reply:
@mansinahar 
I can't push to your remote, but I did go ahead an push that up to beachfront's, so you should be able to pull and merge.

https://github.com/beachfront/prebid-server/tree/beachfront-data-race

Reply:
So in other words, my bad saying LGTM above. I just took a quick look and didn't realize this as a App.